exe-unit/components/gsb-http-proxy/src/counters.rs

```diff
@@ -0,0 +1,183 @@
+use std::sync::{
+    atomic::{AtomicU64, Ordering},
+    Arc, RwLock,
+};
+
+use chrono::{DateTime, Duration, Utc};
+
+use ya_counters::error::CounterError;
+use ya_counters::Result as CountersResult;
+use ya_counters::{Counter, CounterData};
+
+#[derive(Default, Clone, Debug)]
+pub(super) struct Counters {
+    requests: Option<RequestCounter>,
+    requests_duration: Option<SharedRequestsDurationCounter>,
+}
+
+impl Counters {
+    pub(super) fn requests_counter(&mut self) -> impl Counter {
+        self.requests.get_or_insert_with(Default::default).clone()
+    }
+
+    pub(super) fn requests_duration_counter(&mut self) -> impl Counter {
+        self.requests_duration
+            .get_or_insert_with(Default::default)
+            .clone()
+    }
+
+    pub(super) fn on_request(&mut self) -> ResponseHandler {
+        if let Some(counter) = &mut self.requests {
+            counter.on_request();
+        }
+        if let Some(counter) = &mut self.requests_duration {
+            counter.on_request();
+            let counter = Some(counter.clone());
+            return ResponseHandler {
+                counter,
+                ..Default::default()
+            };
```

Comment:
```suggestion
            return ResponseHandler {
                counter: Some(counter.clone()),
                ..Default::default()
            };
```

Reply:
ok (for some reason I just do not like to explicitly type struct field names)